track/1.9 is not up-to-date with the 1.9 bundle

### Bug Description

When we deploy Kubeflow using the terraform plan in this repo in branch `track/1.9`, Charmed Kubeflow UATs are currently failing in Solutions QA at `test_bundle_correctness`.

I believe the issue arises because these terraform plans have not been updated according to the latest Charmed-Kubeflow bundle.

### To Reproduce

1. Deploy the terraform solution in AKS
2. Clone and run the UATs from track/1.9 (with `charmcraft` snap installed)

### Environment

> Let me know if you need this.

### Relevant Log Output

```shell
=========================== short test summary info ============================
FAILED driver/test_kubeflow_workloads.py::test_bundle_correctness - AssertionError: Failed bundle correctness check. Expected: 2.3/* Found: 2.2/stable
assert None
 +  where None = <built-in method match of re.Pattern object at 0x7f393af60450>('2.2/stable')
 +    where <built-in method match of re.Pattern object at 0x7f393af60450> = re.compile('2.3/*').match
 +      where re.compile('2.3/*') = <function compile at 0x7f393f895ea0>('2.3/*')
 +        where <function compile at 0x7f393f895ea0> = re.compile
============= 1 failed, 1 skipped, 1 xfailed in 6003.42s (1:40:03) =============
```

### Additional Context

_No response_

It didn't work.  Got the same error:

```
FAILED driver/test_kubeflow_workloads.py::test_bundle_correctness - AssertionError: Failed bundle correctness check. Expected: 2.3/* Found: 2.2/stable
assert None
 +  where None = <built-in method match of re.Pattern object at 0x7f838cf58450>('2.2/stable')
 +    where <built-in method match of re.Pattern object at 0x7f838cf58450> = re.compile('2.3/*').match
 +      where re.compile('2.3/*') = <function compile at 0x7f83918c9ea0>('2.3/*')
 +        where <function compile at 0x7f83918c9ea0> = re.compile
```

I notice that the only one left in the `applications.tf` with `2.2` is envoy.

--- a/modules/kubeflow/applications.tf
+++ b/modules/kubeflow/applications.tf
@@ -95,7 +95,7 @@ module "katib_ui" {
 }
 
 module "kfp_api" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_api_revision
 }
@@ -115,43 +115,43 @@ module "kfp_db" {
 }
 
 module "kfp_metadata_writer" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_metadata_writer_revision
 }
 
 module "kfp_persistence" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_persistence_revision
 }
 
 module "kfp_profile_controller" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_profile_controller_revision
 }
 
 module "kfp_schedwf" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_schedwf_revision
 }
 
 module "kfp_ui" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_ui_revision
 }
 
 module "kfp_viewer" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_viewer_revision
 }
 
 module "kfp_viz" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=kf-7159-set-channel"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_viz_revision
 }

--- a/modules/kubeflow/applications.tf
+++ b/modules/kubeflow/applications.tf
@@ -95,7 +95,7 @@ module "katib_ui" {
 }
 
 module "kfp_api" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_api_revision
 }
@@ -115,43 +115,43 @@ module "kfp_db" {
 }
 
 module "kfp_metadata_writer" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_metadata_writer_revision
 }
 
 module "kfp_persistence" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_persistence_revision
 }
 
 module "kfp_profile_controller" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_profile_controller_revision
 }
 
 module "kfp_schedwf" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_schedwf_revision
 }
 
 module "kfp_ui" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_ui_revision
 }
 
 module "kfp_viewer" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_viewer_revision
 }
 
 module "kfp_viz" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=track/2.2"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_viz_revision
 }

--- a/modules/kubeflow/applications.tf
+++ b/modules/kubeflow/applications.tf
@@ -97,7 +97,7 @@ module "katib_ui" {
 module "kfp_api" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_api_revision
 }
 
@@ -118,49 +118,49 @@ module "kfp_db" {
 module "kfp_metadata_writer" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_metadata_writer_revision
 }
 
 module "kfp_persistence" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_persistence_revision
 }
 
 module "kfp_profile_controller" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_profile_controller_revision
 }
 
 module "kfp_schedwf" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_schedwf_revision
 }
 
 module "kfp_ui" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_ui_revision
 }
 
 module "kfp_viewer" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_viewer_revision
 }
 
 module "kfp_viz" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel = "2.3/stable"
+  channel    = "2.3/stable"
   revision   = var.kfp_viz_revision
 }
 

--- a/modules/kubeflow/applications.tf
+++ b/modules/kubeflow/applications.tf
@@ -95,7 +95,7 @@ module "katib_ui" {
 }
 
 module "kfp_api" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_api_revision
 }
@@ -115,43 +115,43 @@ module "kfp_db" {
 }
 
 module "kfp_metadata_writer" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_metadata_writer_revision
 }
 
 module "kfp_persistence" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_persistence_revision
 }
 
 module "kfp_profile_controller" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_profile_controller_revision
 }
 
 module "kfp_schedwf" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_schedwf_revision
 }
 
 module "kfp_ui" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_ui_revision
 }
 
 module "kfp_viewer" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_viewer_revision
 }
 
 module "kfp_viz" {
-  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=track/2.3"
+  source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=kf-7159-set-channel"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
   revision   = var.kfp_viz_revision
 }

--- a/modules/kubeflow/applications.tf
+++ b/modules/kubeflow/applications.tf
@@ -97,6 +97,7 @@ module "katib_ui" {
 module "kfp_api" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_api_revision
 }
 
@@ -117,42 +118,49 @@ module "kfp_db" {
 module "kfp_metadata_writer" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_metadata_writer_revision
 }
 
 module "kfp_persistence" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_persistence_revision
 }
 
 module "kfp_profile_controller" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_profile_controller_revision
 }
 
 module "kfp_schedwf" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_schedwf_revision
 }
 
 module "kfp_ui" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_ui_revision
 }
 
 module "kfp_viewer" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_viewer_revision
 }
 
 module "kfp_viz" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
+  channel = "2.3/stable"
   revision   = var.kfp_viz_revision
 }
 

--- a/modules/kubeflow/applications.tf
+++ b/modules/kubeflow/applications.tf
@@ -97,7 +97,6 @@ module "katib_ui" {
 module "kfp_api" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-api/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_api_revision
 }
 
@@ -118,49 +117,42 @@ module "kfp_db" {
 module "kfp_metadata_writer" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-metadata-writer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_metadata_writer_revision
 }
 
 module "kfp_persistence" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-persistence/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_persistence_revision
 }
 
 module "kfp_profile_controller" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-profile-controller/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_profile_controller_revision
 }
 
 module "kfp_schedwf" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-schedwf/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_schedwf_revision
 }
 
 module "kfp_ui" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-ui/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_ui_revision
 }
 
 module "kfp_viewer" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viewer/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_viewer_revision
 }
 
 module "kfp_viz" {
   source     = "git::https://github.com/canonical/kfp-operators//charms/kfp-viz/terraform?ref=track/2.3"
   model_name = var.create_model ? juju_model.kubeflow[0].name : local.model
-  channel    = "2.3/stable"
   revision   = var.kfp_viz_revision
 }
 

